ext/native.go

```diff
@@ -128,20 +128,46 @@ func NativeTypes(args ...any) cel.EnvOption {
 // NativeTypesOption is a functional interface for configuring handling of native types.
 type NativeTypesOption func(*nativeTypeOptions) error
 
+// NativeTypesFieldNameHandler is a handler for mapping a reflect.StructField to a CEL field name.
+// This can be used to override the default Go struct field to CEL field name mapping.
+type NativeTypesFieldNameHandler = func(field reflect.StructField) string
+
+func fieldNameByTag(structTagToParse string) func(field reflect.StructField) string {
```

Comment:
Should this be exported? If it was, we could possibly just drop the `ParseStructTag` option below as it was added recently and there hasn't been a tagged cel-go version since its addition.

Callers who wanted pure tags parsing could then do `ext.ParseStructField(ext.FieldNameByTag)` to achieve the same thing, or they could customize the field parsing further by doing `ext.ParseStructField(myFunc)`.

I didn't do this for now mostly because it means deleting the added `ParseStructTags`. I think there should be no users of this yet though as no new tag published yet...

If the maintainers prefer me to do this just say so and I can update the PR to do so.

Reply:
Specifying a string tag name is likely going to be most common. For more esoteric parsing, I don't mind supporting a separate option that accepts a functional handler as you've set it up currently.